File: Part 02 - Regression/Decision Tree Regression/Demo/decision_tree_regression_demo.R
# read the data
data = read.csv('Position_Salaries.csv')
data = data[2:3]

# create model for train
# install.packages('rpart')
library(rpart)
regressor = rpart(formula = Salary ~ .,
                  data = data,
                  control = rpart.control(minsplit=1))

# predicting a new result
y_pred = predict(regressor, data.frame(Level = 6.5))

# visualize (HR)
# install.packages('ggplot2')
library(ggplot2)
x_grid = seq(min(data$Level), max(data$Level), 0.1)
ggplot() +
  geom_point(aes(x = data$Level, y = data$Salary),
             colour = 'red') +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            colour = 'blue') +
  ggtitle('Truth or Bluff (Decision Tree Regression)') +
  xlab('Level') +
  ylab('Salary')

File: Part 02 - Regression/Decision Tree Regression/Demo/decision_tree_regression_demo.py
# imports
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from sklearn.tree import DecisionTreeRegressor

# read the data and create a dataFrame
data = pd.read_csv('Position_Salaries.csv')  # select file (path)

# separate the data to features and labels
X = data.iloc[:, 1:-1].values  # features - select cols nad rows range
y = data.iloc[:, -1].values  # labels (targets) - select cols nad rows range
# check the separated data
print(f'Features[X]:\n{X}\n{20 * "-"}')
print(f'Labels[y]:\n{y}\n{20 * "-"}')

# create decision tree regression model and train it
tree_regr_model = DecisionTreeRegressor(random_state=0)
tree_regr_model.fit(X, y)

# predict a new value
tree_regr_model.predict([[6.5]])

# visualize the decision tree
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color='red')
plt.plot(X_grid, tree_regr_model.predict(X_grid), color='blue')
plt.title('Truth or Bluff in Decision Tree Regression')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()
